src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -97,7 +97,12 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
 
             if ($overwrite) {
                 if (!is_object($objectOrArray) && !is_array($objectOrArray)) {
-                    throw new UnexpectedTypeException($objectOrArray, 'object or array');
+                    throw new UnexpectedTypeException(sprintf(
+                        'Expected object or array, "%s" found when trying to write property "%s" of path "%s".',
```

Comment:
Please change the message to `PropertyAccessor requires a graph of objects or arrays to operate on, but it found type "%s" while trying to traverse path "%s" at property "%s"`.
Whether it's reading or writing is irrelavant at this point.